File: sdk/AMQP/Node.js/topic/consumer.js
/**
* File: consumer.js
* Description: This is the AMQP consumer handles incoming
*     communication from clients publishing messages to a broker server.
*     Messages can be received over AMQP exchange types including one-to-one,
*     from broadcast pattern, or selectively using specified binding key.
*
* Author: Stanley
* robomq.io (http://www.robomq.io)
*/

var amqp = require("amqp");

var server = "localhost";
var port = 5672;
var vhost = "/";
var username = "guest";
var password = "guest";
var exchangeName = "testEx";
var queueName = "testQ1";
var routingKey = "test.#";

var connection = amqp.createConnection({host: server, port: port, vhost: vhost, login: username, password: password});
connection.on("ready", function(){
	connection.exchange(exchangeName, options = {type: "topic", autoDelete: true, confirm: true}, function(exchange){
		var queue = connection.queue(queueName, options = {exclusive: true, autoDelete: true}, function(queue){
			queue.bind(exchangeName, routingKey, function(){
				queue.subscribe(options = {ack: false}, function(message, headers, deliveryInfo, messageObject){
					//callback funtion on receiving messages
					console.log(message.data.toString());
				});
			});
		});
	}); 
});

File: sdk/SSL/consumer.py
# File: consumer.py
# Description: This is the AMQP SSL consumer handles incoming
#     communication from clients publishing messages to a broker server.
#     This example applies routing-key pattern out of 5 patterns.
#     It needs to acquire the public CA certificate to verify certificate of robomq.io.
#
# Author: Eamin Zhang
# robomq.io (http://www.robomq.io)

import pika
import time

server = "hostname"
port = 5671
vhost = "yourvhost" 
username = "username"
password = "password"
caCert = "http://www.tbs-x509.com/AddTrustExternalCARoot.crt"
exchangeName = "testEx"
queueName = "testQ1"
routingKey = "test"

#callback funtion on receiving messages
def onMessage(channel, method, properties, body):
	print body

while True:
	try:
		#connect
		credentials = pika.PlainCredentials(username, password)
		parameters = pika.ConnectionParameters(host = server, port = port, virtual_host = vhost, credentials = credentials, ssl=True, ssl_options={"ca_certs": caCert})
		connection = pika.BlockingConnection(parameters)
		channel = connection.channel()

		#declare exchange and queue, bind them and consume messages
		channel.exchange_declare(exchange = exchangeName, exchange_type = "direct", auto_delete = True)
		channel.queue_declare(queue = queueName, exclusive = True, auto_delete = True)
		channel.queue_bind(exchange = exchangeName, queue = queueName, routing_key = routingKey)
		channel.basic_consume(consumer_callback = onMessage, queue = queueName, no_ack = True)
		channel.start_consuming()
	except Exception, e:
		#reconnect on exception
		print "Exception handled, reconnecting...\nDetail:\n%s" % e
		try:
			connection.close()
		except:
			pass
		time.sleep(5)